eng/validation/build-arcadewithrepo.ps1

```diff
@@ -0,0 +1,265 @@
+    $headers = @{"Authorization"="Basic $base64AuthInfo"}
+    $content = Invoke-WebRequest -Uri $uri -Headers $headers -ContentType "application/json" -Method Get 
+    return ($content | ConvertFrom-Json).result
+}
+
+function Get-Github-RepoAuthUri($repoName)
+{
+    "https://${githubUser}:${githubPAT}@github.com/${githubOrg}/${repoName}"
+}
+
+function GitHub-Clone($repoName) 
+{
+    $authUri = Get-Github-RepoAuthUri $repoName
+    & git clone $authUri $(Get-Repo-Location $repoName)
+    Push-Location -Path $(Get-Repo-Location $repoName)
+    & git config user.email "${githubUser}@test.com"
+    & git config user.name $githubUser
+    Pop-Location
+}
+
+function Get-Repo-Location($repoName)
+{
+    "$testRoot\$repoName"
+}
+
+function Git-Command($repoName) {
+    Push-Location -Path $(Get-Repo-Location($repoName))
+    try {
+        $gitParams = $args
+        if ($gitParams.GetType().Name -ne "Object[]") {
+            $gitParams = $gitParams.ToString().Split(" ")
+        }
+        Write-Host "Running 'git $gitParams' from $(Get-Location)"
+        $commandOutput = & git @gitParams; if ($LASTEXITCODE -ne 0) { throw "Git exited with exit code: $LASTEXITCODE" } else { $commandOutput }
+        $commandOutput
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+## If able to retrieve a build, get the SHA that it was built from
+$sha = Get-LastKnownGoodBuildSha -azdoOrg $azdoOrg -azdoProject $azdoProject -buildDefinitionId $buildDefinitionId -azdoToken $azdoToken
+
+## Clone the repo from git
+GitHub-Clone $targetRepoName
+
+## Create a branch from the repo with the given SHA. 
+$targetBranch = "dev/" + $githubUser + "/arcade-" + $arcadeSdkVersion
+$githubUri = Get-Github-RepoAuthUri $targetRepoName
+
+$darcBranchName = "refs/heads/" + $targetBranch
+$darcRepoName = "https://github.com/${githubOrg}/${targetRepoName}"
+
+# check to see if branch exists and clean it up if it does
+$branchExists = Git-Command $targetRepoName ls-remote --heads $githubUri refs/heads/$targetBranch
+if($null -ne $branchExists)
+{
+    & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
+    Git-Command $targetRepoName push origin --delete $targetBranch
+}
+Git-Command $targetRepoName checkout -b $targetBranch $sha
+
+## Make the changes to that branch to update Arcade - use darc
+Set-Location $(Get-Repo-Location $targetRepoName)
+& darc update-dependencies --channel ".NET Eng - Validation" --source-repo "arcade" --github-pat $githubPAT --password $barToken
+
+## Push branch to github
+Git-Command $targetRepoName commit -am "Arcade Validation test branch - version ${arcadeSdkVersion}"
+Git-Command $targetRepoName push origin HEAD
+
+## Push branch to AzDO org/project with the official pipeline to build the repo
+## Don't need to do this for Roslyn, but we'll need to do it for Runtime. 
```

Comment:
Not only that, but we should probably not do it in GitHub at all for anything not roslyn.

Reply:
Updated the task list in the other issues for the other repos to not include pushing to GitHub.